src/wrappers/themis/react-native-themis/src/index.tsx

```diff
@@ -3,427 +3,341 @@ import { Buffer } from 'buffer';
+
+    convertInputBase64(context, "context", false);
+    const symmetricKey = convertInputBase64(symmetricKey64, "symmetricKey64", true);
+    const encrypted = convertInputBase64(encrypted64, "encrypted64", true);
+
+    return new Promise((resolve, reject) => {
+        Themis.secureCellContextImprintDecrypt(symmetricKey, encrypted, context, (decrypted: any) => {
+            resolve(Buffer.from(new Uint8Array(decrypted)).toString())
+        }, (error: any) => {
+            reject(error)
+        })
+    });
 }
 
 // secure message sign and verify
 export function secureMessageSign64(
-  plaintext: String,
-  privateKey64: String,
-  publicKey64: String): Promise<string> {
-
-  if (plaintext === "" || plaintext === undefined || plaintext === null) {
-    throw new Error("Parameter plaintext can not be empty");
-  }
-  if (privateKey64 === "" || privateKey64 === undefined || privateKey64 === null) {
-    throw new Error("Parameter privateKey64 can not be empty");
-  }
-
-  const privateKey = Array.from(Buffer.from(privateKey64, 'base64'));
-  const publicKey = publicKey64 !== null && publicKey64 !== "" ?
-    Array.from(Buffer.from(publicKey64, 'base64')) : null;
-
-  return new Promise((resolve, reject) => {
-    Themis.secureMessageSign(plaintext, privateKey, publicKey, (signed: any) => {
-      resolve(Buffer.from(new Uint8Array(signed)).toString("base64"))
-    }, (error: any) => {
-      reject(error)
+    plaintext: String,
+    privateKey64: String): Promise<string> {
+
+    convertInputBase64(plaintext, "plaintext", false);
+    const privateKey = convertInputBase64(privateKey64, "privateKey64", true);
+
+    return new Promise((resolve, reject) => {
+        Themis.secureMessageSign(plaintext, privateKey, (signed: any) => {
+            resolve(Buffer.from(new Uint8Array(signed)).toString("base64"))
+        }, (error: any) => {
+            reject(error)
+        })
     })
-  })
 }
 
 export function secureMessageVerify64(
-  signed64: String,
-  privateKey64: String,
-  publicKey64: String): Promise<string> {
-
-  if (signed64 === "" || signed64 === undefined || signed64 === null) {
-    throw new Error("Parameter signed64 can not be empty");
-  }
-  if (publicKey64 === "" || publicKey64 === undefined || publicKey64 === null) {
-    throw new Error("Parameter publicKey64 can not be empty");
-  }
-
-  const signed = Array.from(Buffer.from(signed64, 'base64'));
-  const privateKey = privateKey64 !== null && privateKey64 !== "" ?
-    Array.from(Buffer.from(privateKey64, 'base64')) : null;
-  const publicKey = Array.from(Buffer.from(publicKey64, 'base64'));
-
-  return new Promise((resolve, reject) => {
-    Themis.secureMessageVerify(signed, privateKey, publicKey, (verified: any) => {
-      resolve(Buffer.from(new Uint8Array(verified)).toString())
-    }, (error: any) => {
-      reject(error)
+    signed64: String,
+    _privateKey64: String = "",
+    publicKey64: String): Promise<string> {
```

Comment:
Do optional parameters actually work like that in TypeScript? You can make a parameter *in the middle* of the list optional? 🤔

Reply:
To make it back compatible I had to return the unused privateKey64 parameter to the middle of the list of parameters. But typescript wants to use the required parameter in the function body. So, to omit the error I had to add undersign character. That indicates the parameter is maybe unused in the future.

Reply:
Nah, I'm just surprised it allows to put default values in the middle of the list. Usually it's allowed only at the end, as that makes it easy to disambiguate which parameters should be defaulted. However, maybe the compiler is smart enough, since if there is only one optional parameter, it's unambiguous. But I'm not so sure about vanilla JS.

Could you please add a test to ensure that both 2-argument and 3-argument calls keep working?